relayer/src/cli/programs/prove-op-message.ts

```diff
@@ -0,0 +1,50 @@
+import { getNetworkId } from "../../common/network";
+import { OptimismRelayerService } from "../../services/";
+
+export type ProgramOptions = {
+  l1RpcUrl: string;
+  l2RpcUrl: string;
+  walletPrivateKey: string;
+  l2TransactionHash: string;
+  customNetwork?: string;
+};
+
+export class OPMessageProverProgram {
+  constructor(
+    readonly l2RelayerFrom: OptimismRelayerService,
+    readonly l2TransactionHash: string,
+  ) {}
+
+  static validateArgs(args: ProgramOptions): Array<string> {
+    const validationErrors = [];
+
+    if (!args.l2TransactionHash.startsWith("0x") || args.l2TransactionHash.trim().length != 66) {
+      validationErrors.push(
+        `Invalid argument value for "--l2-transaction-hash": "${args.l2TransactionHash}" doesn't look like a txn hash...`,
+      );
+    }
+
+    return validationErrors;
+  }
+
+  static async createFromArgs(options: ProgramOptions): Promise<OPMessageProverProgram> {
+    const l2ChainId = await getNetworkId(options.l2RpcUrl);
+    const l1ChainId = await getNetworkId(options.l1RpcUrl);
+
+    const OPRelayer = new OptimismRelayerService(
+      options.l1RpcUrl,
+      options.l2RpcUrl,
+      options.walletPrivateKey,
+      l1ChainId,
+      l2ChainId,
+      undefined,
```

Comment:
I really don't like that it is possible to pass undefined here. The constructor actually requires this parameter. 
We end up with the following: The `destinationChainId` of the `OptimismRelayerService` is declared as number (without the `?`-optional declaration), but in some cases it might be undefined. This can lead to runtime errors, which we actually try to prevent and therefore use TypeScript. Sure it might not be a problem at the moment because `destinationChainId` is only used by the ethereum relayer, but such things will bring problems in the future. You can see this with the l1 tx hash problem that we have with polygon now.

This is only possible because we don't use TypeScript's strict mode in the relayer, yet. I actually enabled it in my architectural changes for the steps. 
As we want to get this merged quickly don't enable the strict mode now, because there will be more typings that need to be solved. But it should be enabled as soon as possible.

This parameter should still be changed. Either provide a value here or make the property optional and handle the case it is undefined in all places we use the `destinationChainId`. Since it is only used in the ethereum relayer, maybe there is also another way to get it without requiring the `BaseRelayerService` to have it as a property.

Sorry for the long text, I know you quickly add such an unwanted `undefined` if your editor doesn't tell you, because strict mode is not enabled. It's just an example of how bad the relayer is in shape, because it evolved from a simple script.

Reply:
Well, you missed it because the strict mode is not enabled ;)
Enabling the typescript strict mode is quite important I think, regardless of my step architecture change. That's all I wanted to say.